components/widgets/inputs/users_emails_input.jsx

```diff
@@ -94,6 +94,24 @@ export default class UsersEmailsInput extends React.Component {
         return user.id || user.value;
     }
 
+    getOptionLabel = (user) => {
+        if (user.value && isEmail(user.value)) {
+            return user.value;
+        }
+
+        const parts = getLongDisplayNameParts(user);
+        let fullName = null;
+        if (parts.fullName) {
+            fullName = parts.fullName;
+        }
+        let nickname = null;
+        if (parts.nickname) {
+            nickname = parts.nickname;
+        }
+
+        return `${parts.displayName || ''} ${fullName || ''} ${nickname || ''}`;
```

Comment:
I don't think the blank spaces are needed if elements are missing, right?
```suggestion
        const results = [];
        if (parts.displayName) {
             results.push(parts.displayName);
        }
        if (parts.fullName) {
            results.push(parts.fullName);
        }
        if (parts.nickname) {
            results.push(parts.nickname);
        }
    return results.join(' ');
```

Reply:
for the context is not specially important, but I think you are right, is more correct for sure, I'll apply your proposed changes.